tracer/src/Datadog.Trace/SpanLink.cs

```diff
@@ -0,0 +1,44 @@
+// <copyright file="SpanLink.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.Propagators;
+
+namespace Datadog.Trace;
+
+/// <summary>
+/// The SpanLink contains the information needed for a decroated span for its Span Links.
+/// </summary>
+internal class SpanLink
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SpanLink"/> class.
+    /// A span link describes a tuple of trace id and span id
+    /// in OpenTelemetry that's called a Span Context, which may also include tracestate and trace flags.
+    /// </summary>
+    /// <param name="spanLinkContext">The context of the spanlink to extract attributes from</param>
+    /// <param name="optionalAttributes">Optional dictionary of attributes to take for the spanlink.</param>
+    internal SpanLink(SpanContext spanLinkContext, Dictionary<string, object> optionalAttributes)
+    {
+        SpanLinkContext = spanLinkContext;
+        if (optionalAttributes is not null)
+        {
+            Attributes = optionalAttributes;
+        }
+    }
+
+    internal SpanLink(Span spanToLink, Dictionary<string, object> optionalAttributes)
+        : this(spanToLink.Context, optionalAttributes)
+    {
+    }
+
+    internal Dictionary<string, object> Attributes { get;  }
+
+    internal SpanContext SpanLinkContext { get;  }
```

Comment:
[naming nit pick] We can probably call this property `SpanContext` or even `Context`. No need for `SpanLink` in the name since it's already a member of the `SpanLink` type.

Reply:
went with context to avoid the type + property having the same name